src/main/java/org/springframework/data/jpa/repository/query/StoredProcedureAttributeSource.java

```diff
@@ -100,36 +101,28 @@ private String deriveProcedureNameFrom(Method method, Procedure procedure) {
 	private StoredProcedureAttributes newProcedureAttributesFrom(Method method,
 			NamedStoredProcedureQuery namedStoredProc, Procedure procedure) {
 
-		String outputParameterName = null;
-		Class<?> outputParameterType = null;
+		List<String> outputParameterNames = new ArrayList<>();
+		List<Class<?>> outputParameterTypes = new ArrayList<>();
 
 		if (!procedure.outputParameterName().isEmpty()) {
-
 			// we give the output parameter definition from the @Procedure annotation precedence
-			outputParameterName = procedure.outputParameterName();
+			outputParameterNames.add(procedure.outputParameterName());
 		} else {
 
 			// try to discover the output parameter
 			List<StoredProcedureParameter> outputParameters = extractOutputParametersFrom(namedStoredProc);
 
-			if (outputParameters.size() != 1 && !void.class.equals(method.getReturnType())) {
-				throw new IllegalStateException(String.format(
-						"Could not create ProcedureAttributes from %s. We currently support exactly one output parameter!", method));
-			}
-
-			if (!outputParameters.isEmpty()) {
-				StoredProcedureParameter outputParameter = outputParameters.get(0);
-				outputParameterName = outputParameter.name();
-				outputParameterType = outputParameter.type();
+			for (StoredProcedureParameter outputParameter : outputParameters) {
+				outputParameterNames.add(outputParameter.name());
+				outputParameterTypes.add(outputParameter.type());
 			}
 		}
 
-		if (outputParameterType == null || Object.class.equals(outputParameterType)
-				|| void.class.equals(outputParameterType)) {
-			outputParameterType = method.getReturnType();
+		if (outputParameterTypes.isEmpty()) {
```

Comment:
I don't think this equivalent to what we had before.
The old version was basically: If we don't have a return type to deduce from the procedure definition, we use the one from the method.

Reply:
I think that is basically the same. Types is just now a list instead of a single variable